syncer/syncer.go

```diff
@@ -1675,11 +1676,19 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext) e
 			s.tctx.L().Info("replace ddls to preset ddls by sql operator in normal mode", zap.String("event", "query"), zap.Strings("preset ddls", appliedSQLs), zap.Strings("ddls", needHandleDDLs), zap.ByteString("raw statement", ev.Query), log.WrapStringerField("position", ec.currentPos))
 			needHandleDDLs = appliedSQLs // maybe nil
 		}
+
 		job := newDDLJob(nil, needHandleDDLs, *ec.lastPos, *ec.currentPos, nil, nil, *ec.traceID)
 		err = s.addJobFunc(job)
 		if err != nil {
 			return err
 		}
+
+		// when add ddl job, will execute ddl and then flush checkpoint.
+		// if execute ddl failed, the execErrorDetected will be true.
+		if s.execErrorDetected.Get() {
+			return terror.ErrSyncerUnitHandleDDLFailed.Generate()
```

Comment:
1. How about including more information about this DDL job (at least the DDL statement)?
2. One more question is there any problem when resuming from an older checkpoint before any online DDL operation, like:
```
1. some online DDL operation
2. save it into TiDB (https://github.com/pingcap/dm/blob/8182696221688b61571bca4febd7d5df17194cb0/syncer/online_ddl.go#L205-L207)
3. error occurred
4. resume from any position older than step-1.
```
Is it possible `append`ed multiple times in step-2? 😂

Reply:
1. add ddl query in error in 92ede5b
2. judge ddl before save it in 3c602b0